### PR TITLE
Comics from Heroes Cached in Database

### PR DESCRIPTION
**Pull request by joseantoniocampillo**
One simple way for cached heros in DB to get offline after load

### CODE REVIEW

File: app/src/main/java/com/architectcoders/arquitectomarvel/data/database/MarvelDao.kt [hunk 1]
```diff
@@ -19,28 +21,40 @@ interface MarvelDao {
...
 
     @Query("DELETE FROM characterentity")
     suspend fun deleteAllLocalCharacters()
+
+    // Comics
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertComics(comicEntity: List<ComicEntity>)
+
+    @Query("DELETE FROM COMICENTITY WHERE idHero = :idHero")
+    suspend fun deleteComicsForHero(idHero: Int)
```
Comment (juancoob) [inline app/src/main/java/com/architectcoders/arquitectomarvel/data/database/MarvelDao.kt:42]: @joseantoniocampillo could you change "hero" words to "character"?

File: app/src/main/java/com/architectcoders/arquitectomarvel/data/database/MarvelDao.kt [hunk 2]
```diff
@@ -19,28 +21,40 @@ interface MarvelDao {
...
+
+    @Query("SELECT * FROM ComicEntity WHERE idHero = :idHero")
+    fun selecetComicsForHero(idHero: Int): Flow<List<ComicEntity>>
```
Comment (juancoob) [inline app/src/main/java/com/architectcoders/arquitectomarvel/data/database/MarvelDao.kt:45]: @joseantoniocampillo Please, fix the typo "selecet"

File: app/src/main/java/com/architectcoders/arquitectomarvel/data/database/RoomDataSource.kt [hunk 3]
```diff
@@ -2,43 +2,50 @@ package com.architectcoders.arquitectomarvel.data.database
...
-    override suspend fun deleteLocalFavoriteComic(favoriteComic: ComicResult) =
-        characterDao.deleteLocalFavoriteComic(favoriteComic.toFavoriteComicEntity)
+    override suspend fun deleteLocalFavoriteComic(idHero: Int) =
+        marvelDao.deleteComicsForHero(idHero)
 
     override fun getPagingSourceFromCharacterEntity(): PagingSource<Int, CharacterEntity> =
-        characterDao.getPagingSourceFromCharacterEntity()
+        marvelDao.getPagingSourceFromCharacterEntity()
+
+    override suspend fun fetchComicsForHero(map: Map<String, Any>) {
```
Comment (juancoob) [inline app/src/main/java/com/architectcoders/arquitectomarvel/data/database/RoomDataSource.kt:38]: @joseantoniocampillo could you remove the brackets and add the "=" for both functions?

**Review by juancoob — APPROVED** [review]
Thanks for the effort @joseantoniocampillo!! It is similar to RemoteMediator, but in my opinion, It doesn't work as smooth as the last one.

**Review by JorgeGreek — APPROVED** [review]
ok

**Pull request merged**